src/test/java/com/google/firebase/auth/FirebaseAuthIT.java

```diff
@@ -140,37 +141,98 @@ public void testDeleteNonExistingUser() throws Exception {
+      checkRecreate(user.getUid());
     } finally {
-      auth.deleteUserAsync(userRecord.getUid()).get();
+      auth.deleteUserAsync(randomUser.uid).get();
+    }
+  }
+
+  @Test
+  public void testLookupUserByPhone() throws Exception {
+    RandomUser randomUser1 = RandomUser.create();
+    String phoneNumber1 = null;
+    RandomUser randomUser2 = RandomUser.create();
+    String randomPhoneNumber2 = randomPhoneNumber();
+    RandomUser randomUser3 = RandomUser.create();
+    String randomPhoneNumber3 = randomPhoneNumber();
+    try {
+      UserRecord user1 = createUser(
+          randomUser1.uid, /* phoneNumber= */ null, randomUser1.email);
+      UserRecord user2 = createUser(
+          randomUser2.uid, randomPhoneNumber2, randomUser2.email);
+      UserImportResult user3 = importUser(
+          randomUser3.uid, randomPhoneNumber3, randomUser3.email, "google.com");
+
+      UserRecord lookedUpRecord = auth.getUserByPhoneNumberAsync(randomPhoneNumber2).get();
+      assertEquals(lookedUpRecord.getUid(), randomUser2.uid);
+
+      lookedUpRecord = auth.getUserByPhoneNumberAsync(randomPhoneNumber3).get();
+      assertEquals(lookedUpRecord.getUid(), randomUser3.uid);
+    } finally {
+      auth.deleteUserAsync(randomUser1.uid).get();
+      auth.deleteUserAsync(randomUser2.uid).get();
+      auth.deleteUserAsync(randomUser3.uid).get();
+    }
+  }
+
+  @Test
+  public void testLookupUserByFederatedId() throws Exception {
+    RandomUser randomUser1 = RandomUser.create();
+    String phoneNumber1 = null;
+    RandomUser randomUser2 = RandomUser.create();
+    String randomPhoneNumber2 = randomPhoneNumber();
+    RandomUser randomUser3 = RandomUser.create();
+    String randomPhoneNumber3 = randomPhoneNumber();
+    try {
+      UserRecord user1 = createUser(
+          randomUser1.uid, /* phoneNumber= */ null, randomUser1.email);
+      UserRecord user2 = createUser(
+          randomUser2.uid, randomPhoneNumber2, randomUser2.email);
+      UserImportResult user3 = importUser(
+          randomUser3.uid, randomPhoneNumber3, randomUser3.email, "google.com");
+
+      UserRecord lookedUpRecord = auth.getUserByFederatedIdAsync(
+          randomUser3.uid + "_google.com", "google.com").get();
+      assertEquals(lookedUpRecord.getUid(), randomUser3.uid);
+      assertEquals(2, lookedUpRecord.getProviderData().length);
+      List<String> providers = new ArrayList<>();
+      for (UserInfo provider : lookedUpRecord.getProviderData()) {
+        providers.add(provider.getProviderId());
+      }
+      assertTrue(providers.contains("phone"));
+      assertTrue(providers.contains("google.com"));
+
+      try {
+        // Verify that lookup by federated identifier does not accept "phone".
+        lookedUpRecord = auth.getUserByFederatedIdAsync(
+            randomPhoneNumber3, "phone").get();
+        fail("No error thrown for non-federated provider");
+      } catch (IllegalArgumentException ignored) {
+        // expected
```

Comment:
nit: You could rename the exception s/ignored/expected and then get rid of the comment.
http://go/java-style#s6.2-caught-exceptions